python/astro_metadata_translator/translators/decam.py

```diff
@@ -248,3 +251,44 @@ def to_detector_name(self):
         # Docstring will be inherited. Property defined in properties.py
         name = self.to_detector_unique_name()
         return name[1:]
+
+    @classmethod
+    def fix_header(cls, header, obsid, filename=None):
+        """Fix DECam headers.
+
+        Parameters
+        ----------
+        header : `dict`
+            The header to update.  Updates are in place.
+        obsid : `str`
+            Unique observation identifier associated with this header.
+            Will always be provided.
+        filename : `str`, optional
+            Filename associated with this header. May not be set since headers
+            can be fixed independently of any filename being known.
+
+        Returns
+        -------
+        modified = `bool`
+            Returns `True` if the header was updated.
+
+        Notes
+        -----
+        Fixes the following issues:
+        * If OBSTYPE contains "zero" or "bias",
+          update the FILTER keyword to "solid plate 0.0 0.0".
+
+        Corrections are reported as debug level log messages.
+        """
+        modified = False
+
+        obsid = header.get("OBSID", "unknown")
```

Comment:
Do not recalculate obsid since it's passed into the function.

What you need to do is instead:

```python
        # Calculate the standard label to use for log messages
        log_label = cls._construct_log_prefix(obsid, filename)
```
and use that log_label in the log message below instead of the `obsid`.